SDWebImage/SDWebImagePrefetcher.m

```diff
@@ -103,12 +105,12 @@ - (nullable SDWebImagePrefetchToken *)prefetchURLs:(nullable NSArray<NSURL *> *)
 }
 
 - (void)startPrefetchWithToken:(SDWebImagePrefetchToken * _Nonnull)token {
-    NSPointerArray *operations = token.operations;
+    NSPointerArray *operations = token.loadOperations;
     for (NSURL *url in token.urls) {
         __weak typeof(self) wself = self;
         SDAsyncBlockOperation *prefetchOperation = [SDAsyncBlockOperation blockOperationWithBlock:^(SDAsyncBlockOperation * _Nonnull asyncOperation) {
              __strong typeof(wself) strongSelf = wself;
-             if (!strongSelf) {
+             if (!strongSelf || asyncOperation.isCancelled) {
                  return;
              }
              id<SDWebImageOperation> operation = [strongSelf.manager loadImageWithURL:url options:strongSelf.options context:strongSelf.context progress:nil completed:^(UIImage * _Nullable image, NSData * _Nullable data, NSError * _Nullable error, SDImageCacheType cacheType, BOOL finished, NSURL * _Nullable imageURL) {
```

Comment:
@dreampiggy Another potential issue is here, `loadImageWithURL`'s return type marked `nullable`, I checked the code, currently it would not return `nil`, but let's assume it would return `nil` some day in the future, if this situation exist, we need to do some additional processes.

```
- (nullable SDWebImageCombinedOperation *)loadImageWithURL:(nullable NSURL *)url
                                                   options:(SDWebImageOptions)options
                                                  progress:(nullable SDImageLoaderProgressBlock)progressBlock
                                                 completed:(nonnull SDInternalCompletionBlock)completedBlock;
```

Reply:
If this should never return nil, the correct way shoud be changing the public API to nonnull.

Reply:
During the 5.0 migration, I’ve already fix some nullability issue, like the some delegate method of `NSURL`, which should actually be `nonnill`. I think this may because of historical issue, at that time, there are so many API need to mark nullability, so they placed nullable for those API they can not ensure nonnill.

Reply:
Maybe needs to fix those `nullable` issues in 6.0. Now, let's just add an assert to check.